### PR TITLE
Added reasons as to why the bus was deprecated.

### PR DESCRIPTION
**Pull request by indualagarsamy**
Fixes: https://github.com/Particular/docs.particular.net/issues/1327

### CODE REVIEW

**Comment by indualagarsamy** [comment]
@boblangley @Particular/docs-maintainers - please review

File: nservicebus/upgrades/5to6-moving-away-from-ibus.md [hunk 1]
```diff
@@ -7,28 +7,38 @@ Starting with Version 6, the `IBus` interface has been deprecated.
 
 The `IBus` interface was one of the key interfaces when using previous versions of NServiceBus. It provided access to many of the operations like sending messages, subscribing to events, and manipulating headers. The `IBus` interface was available through dependency injection in message handlers, any custom components that were registered in the container and through other mechanisms like the saga base class.
 
+
+## Why was IBus deprecated?
```
Comment (SimonCropp) [inline nservicebus/upgrades/5to6-moving-away-from-ibus.md]: i know this approach is in other places but i dislike rhetorical questions as titles. how about "Reasons for IBus deprecated"

File: nservicebus/upgrades/5to6-moving-away-from-ibus.md [hunk 2]
```diff
@@ -7,28 +7,38 @@ Starting with Version 6, the `IBus` interface has been deprecated.
...
+
+The problem with `IBus` is that the interface contained numerous methods and properties. However, not all of them are valid in certain scenarios. For example, methods like `Reply` and `ForwardCurrentMessageTo` are always available on the `IBus` interface, but it only makes sense to use those in the context of handling an incoming message. Using them in other scenarios will throw an exception.
```
Comment (SimonCropp) [inline nservicebus/upgrades/5to6-moving-away-from-ibus.md]: > The `IBus` interface contained numerous methods and properties. However, not all of them are valid in all scenarios where an instance of `IBus` was exposed. This lead to confusion over which API were valid or recommended in these scenarios. For example, the methods `Reply` and `ForwardCurrentMessageTo` are always available on the `IBus` interface, but it only makes sense to use those in the context of handling an incoming message. Using them in other scenarios will throw an exception.

File: nservicebus/upgrades/5to6-moving-away-from-ibus.md [hunk 3]
```diff
@@ -8,11 +8,11 @@ Starting with Version 6, the `IBus` interface has been deprecated.
...
 
 
-## Why was IBus deprecated?
+## Reasons for deprecating IBus
 
-The problem with `IBus` is that the interface contained numerous methods and properties. However, not all of them are valid in certain scenarios. For example, methods like `Reply` and `ForwardCurrentMessageTo` are always available on the `IBus` interface, but it only makes sense to use those in the context of handling an incoming message. Using them in other scenarios will throw an exception.
+The `IBus` interface contained numerous methods and properties. However, not all of them are valid in all scenarios where an instance of `IBus` was exposed and this can lead to confusion. For example, the methods `Reply` and `ForwardCurrentMessageTo` are always available on the `IBus` interface, but it is only relevant to use it in the context of handling an incoming message. Using them in other scenarios will throw an exception.
```
Comment (boblangley) [inline nservicebus/upgrades/5to6-moving-away-from-ibus.md]: >but it is only relevant to use it in the context of handling an incoming message.

The subject are the methods not the bus, I would change to:

>but they are only relevant to use in the context of handling an incoming message.

Switching from **it** to **they** also ties this sentence better into the following one.

Comment (indualagarsamy) [inline nservicebus/upgrades/5to6-moving-away-from-ibus.md]: yep. thanks. done.

File: nservicebus/upgrades/5to6-moving-away-from-ibus.md [hunk 4]
```diff
@@ -7,28 +7,38 @@ Starting with Version 6, the `IBus` interface has been deprecated.
...
 
+
+## Reasons for deprecating IBus
+
+The `IBus` interface contained numerous methods and properties. However, not all of them are valid in all scenarios where an instance of `IBus` was exposed and this can lead to confusion. For example, the methods `Reply` and `ForwardCurrentMessageTo` are always available on the `IBus` interface, but they are only relevant to use it in the context of handling an incoming message. Using them in other scenarios will throw an exception.
+
+For users who are new to the NServiceBus API, the fact that `IBus` was available in message handlers using Dependency Injection wasn't obvious, especially when trying to send or publish messages from inside message handlers. Rather than using dependency injection, a `IMessageHandlerContext` parameter is now available in all the message handlers. This parameter will expose all of the appropriate actions to the message handler. Methods that aren't applicable will not be available making the API simpler.
```
Comment (danielmarbach) [inline nservicebus/upgrades/5to6-moving-away-from-ibus.md]: There is also a technical reason. Previously IBud impl was bound to a per request scope which sometimes was implemented as PerThreadScope. This doesn't work with async see my floating state post. Should we mention it?

**Pull request merged**